server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -480,10 +479,27 @@ public static class Clusters implements ToXContentFragment, Writeable {
         private final int successful;
         private final int skipped;
 
+        /**
+         * Creates a Clusters object in its "final" state, where successful + skipped = total.
+         * All clusters must be accounted for.
+         * @param total total number of clusters being searched
+         * @param successful
+         * @param skipped
+         */
         public Clusters(int total, int successful, int skipped) {
+            this(total, successful, skipped, true);
+        }
+
+        /**
+         * If finalState = true, creates a Clusters object in its "final" state, where successful + skipped = total
+         * If finalState = false, then the above condition is not enforced. This would happen in a case
+         * where an async search is happening and the Clusters state will be incrementally updated
```

Comment:
I keep forgetting: this is only true for async search, or is it the same also for synchronous search when minimizing roundtrips?

Reply:
Synchronous search will also use it because of the way that Async Search just layers on top of Sync Search - the logic for creating this new Cluster object is in TransportSearchAction which doesn't check whether it is an async search or not.